src/tests/ftest/interoperability/updown_grade.py

```diff
@@ -0,0 +1,353 @@
+        """
+        decoded = {}
+        for key, val in outdata.items():
+            if isinstance(val, bytes):
+                decoded[key.decode()] = val
+            else:
+                decoded[key] = base64.b64decode(val)
+
+        self.log.info("Verifying get_attr output:")
+        self.log.info("  get_attr data: %s", indata)
+        self.log.info("  set_attr data: %s", decoded)
+
+        for attr, value in indata.items():
+            if value != decoded.get(attr.decode(), None):
+                self.fail(
+                    "FAIL: Value does not match after get({}), Expected "
+                    "val={} and received val={}".format(attr, value,
+                                        decoded.get(attr.decode(), None)))
+
+    def check_result(self, result):
+        """check for command result, raise failure when error cncountered
+
+        Args:
+             result (dict): dictionary of result to check.
+        """
+        self.log.info("--check_result, result= %s", result)
+        if result[0]['exit_status'] != 0:
+            self.fail("##Error detected from check_result")
+        else:
+            self.log.info("--check_result passed")
+
+    def show_daos_version(self, all_hosts, hosts_client):
+        """show daos version
+
+        Args:
+            all_hosts (list): all hosts.
+            hosts_client (list): client hosts to show daos and dmg version.
+        """
+        result = run_pcmd(all_hosts, "rpm -qa | grep daos")
+        self.check_result(result)
+        result = run_pcmd(hosts_client, "dmg version")
+        self.check_result(result)
+        result = run_pcmd(hosts_client, "daos version")
+        self.check_result(result)
+
+    def updowngrade(self, hosts, updown, rpms):
+        """Upgrade downgrade hosts
+
+        Args:
+            hosts (list): test hosts.
+            updown (str): upgrade or downgrade
+            rpms (list): full path of RPMs to be upgrade or downgrade
+        """
+        cmds = []
+        for rpm in rpms:
+            cmds.append("sudo yum {} -y {}".format(updown, rpm))
+        cmds.append("sudo ipcrm -a")
+        cmds.append("sudo ipcs")
+        self.log.info("==upgrade_downgrading on hosts: %s", hosts)
+        for cmd in cmds:
+            self.log.info("==cmd= %s", cmd)
+            result = run_pcmd(hosts, cmd)
+        self.log.info("==sleeping 5 more seconds")
+        time.sleep(5)
+        self.log.info("==pcmd yum upgrade/downgrade result= %s", result)
+
+    def daos_ver_after_upgraded(self, host):
+        """To display daos and dmg version, and check for error.
+
+        Args:
+            host (str): test host.
+        """
+        cmds = [
+                "daos version",
+                "dmg version",
+                "daos pool query {}".format(self.pool.identifier)
+               ]
+        for cmd in cmds:
+            self.log.info("==cmd= %s", cmd)
+            result = pcmd(host, cmd, False)
```

Comment:
Can't we just use our usual `get_dmg_command().version()`? The version command should work even if the server isn't running.